packages/fast-usdc-contract/src/exos/settler.ts

```diff
@@ -422,15 +448,37 @@ export const prepareSettler = (
           })();
           if (!dest) return;
 
-          const { reference } = parseAccountId(dest);
+          const { namespace, reference } = parseAccountId(dest);
           const amt = AmountMath.make(USDC, fullValue);
-          const txfrOrSendV =
-            reference === currentChainRef
-              ? E(settlementAccount).send(dest, amt)
-              : E(settlementAccount).transfer(dest, amt, {
-                  forwardOpts: { intermediateRecipient },
-                });
-          void vowTools.watch(txfrOrSendV, this.facets.transferHandler, txHash);
+
+          if (namespace === 'cosmos') {
+            const txfrOrSendV =
+              reference === currentChainRef
+                ? E(settlementAccount).send(dest, amt)
+                : E(settlementAccount).transfer(dest, amt, {
+                    forwardOpts: { intermediateRecipient },
+                  });
+            void vowTools.watch(
+              txfrOrSendV,
+              this.facets.transferHandler,
+              txHash,
+            );
+          } else if (isCctpDestSupported(dest)) {
+            // send to Noble then call `.depositForBurn(dest, amt)`
+            void vowTools.watch(
+              E(settlementAccount).transfer(intermediateRecipient, amt),
+              this.facets.intermediateTransferHandler,
+              { amt, dest, txHash },
+            );
+          } else {
+            log(
+              '🚨 forward not attempted!',
+              'unsupported destination',
```

Comment:
I suppose it's worth double-checking for unsupported destination here, but this isn't actually reachable in production, right? We detect unsupported destination as soon as we decode the EUD, right?

Reply:
> We detect unsupported destination as soon as we decode the EUD, right?

Yes and this results in `AdvanceSkipped`. When the mint eventually arrives, a `.forward()` is attempted for this TX.

So it is possible in production, just unlikely. _Unlikely if they are using the official front end and not trying to interact programmatically._